src/lib/profiles/data-management/Current/LoggingManagement.h

```diff
@@ -146,6 +146,36 @@ enum LoggingManagementStates
 // forward class declaration
 class LogBDXUpload;
 
+/**
+ * @brief
+ *   A helper class used in initializing logging management.
+ *
+ * The class is used to encapsulate the resources allocated by the caller and denotes
+ * resources to be used in logging events of a particular importance.  Note that
+ * while resources referring to the counters are used exclusively by the
+ * particular importance level, the buffers are shared between `this` importance
+ * level and events that are "more" important.
+ */
+
+struct LogStorageResources
+{
+    constexpr LogStorageResources(void * inBuffer, size_t inBufferSize, nl::Weave::Platform::PersistedStorage::Key * inCounterKey,
+                                  uint32_t inCounterEpoch, nl::Weave::PersistedCounter * inCounterStorage) :
+        mBuffer(inBuffer),
+        mBufferSize(inBufferSize), mCounterKey(inCounterKey), mCounterEpoch(inCounterEpoch), mCounterStorage(inCounterStorage) { };
+    void * mBuffer;     //< Buffer to be used as a storage at the particular importance level and shared with more important events.
+                        //Must not be NULL.  Must be large enough to accomodate the largest event emitted by the system.
```

Comment:
Do we have tooling to check what the largest event emitted by the system might be or an assertion that fires when this "largest event emitted..." fails?

Reply:
We do not have tooling that would check that at compile time (not sure even if it is possible). For runtime checks -- we will fail gracefully if the event is too large to fit into the first buffer, we don't handle the case where the event is too large to fit into subsequent buffers reserved for higher importance levels.  I see a way to add that check relatively easily, will add that check.

Reply:
Nevermind, I see I already added that check some time ago